server/bin/test_data.json

```diff
@@ -27,57 +93,59 @@
             "type": "yum",
             "vendor": "test-vendor",
             "content_url": "/foo/path",
-            "gpg_url": "/foo/path/gpg/",
```

Comment:
What's driving the removal of these? I'm somewhat concerned what impact this will have on some of QE's test suite. Is it possible to leave these in place without breaking the RPM building bits? Perhaps using --nogpgcheck?

Reply:
Automatic signing of generated RPMs is very complicated, because GPG requires using passhrase (see my comment in commit message). Thus I decided to remove `gpg_url` from `test_data.json`

I can create RPMs without GPG signs (it is default behavior of rpmbuild). The problem is on different place. This 'gpg_url' influences definition of repository in `/etc/yum.repos.d/redhat.repo` created automatically by e.g. subscription-manager plugin on the registered system. The `/etc/yum.repos.d/redhat.repo` is automatically generated from entitlement certificates installed in `/etc/pki/entitlement` (you can see content of cert usint `rct cc cert.pem`). The entitlement certificate is generated by candlepin server, when system attaches som subscription using e.g.: `subscription-manager attach --pool <pool_id>`

When definition repository in `redhat.repo` does include not empty string of GPG and rpm is not signed, then yum/dnf/PackageKit/microdnf will not allow to install such RPM to the system.

I will discuss this with QE team.

Reply:
Would it be possible to have values here that are valid enough to not break the GPG signing, but still have values present for testing?

Reply:
I think that will not be possible. I will try to do more experiments with signing RPMs using GPG key.